File: P4/ruby/Loot.rb
# encoding: UTF-8

require_relative 'LootToUI'

module Deepspace
	class Loot
		def initialize (nSu, nWe, nSh, nHa, nMe, ef=false, city=false) # Loot(nsu : int, nw : int, nsh : int, nh : int, nm : int, ef : boolean, city : boolean)
			@nSupplies = nSu
			@nWeapons = nWe
			@nShields = nSh
			@nHangars = nHa
			@nMedals = nMe

			@efficient = ef
			@spaceCity = city
		end
		
		attr_reader :nSupplies, :nWeapons, :nShields, :nHangars, :nMedals, :efficient, :spaceCity

		def self.newCopy (l)
			new(l.nSupplies, l.nWeapons, l.nShields, l.nHangars, l.nMedals, l.efficient, l.spaceCity)
		end

		def getUIversion
			LootToUI.new(self)
		end

		def to_s
			getUIversion.to_s
		end
		
	end # class
end # module

if $0 == __FILE__
	puts Deepspace::Loot.new(1, 2, 3, 4, 5)
	puts Deepspace::Loot.new(1, 2, 3, 4, 5).getUIversion
end

File: P4/java/deepspace/src/deepspace/Loot.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

class Loot {
    private int nSupplies, nWeapons, nShields, nHangars, nMedals;
    private boolean efficient, spaceCity;
    
    Loot (int nSu, int nWe, int nSh, int nHa, int nMe) {
        nSupplies = nSu;
        nWeapons = nWe;
        nShields = nSh;
        nHangars = nHa;
        nMedals = nMe;
        
        efficient = false;
        spaceCity = false;
    }
    
    Loot (int nSu, int nWe, int nSh, int nHa, int nMe, boolean ef, boolean city) {
        nSupplies = nSu;
        nWeapons = nWe;
        nShields = nSh;
        nHangars = nHa;
        nMedals = nMe;
        
        efficient = ef;
        spaceCity = city;
    }
    
    Loot(Loot l){ //Hago este constructor de copia para que en la clase EnemyStarShip no se creen dos objetos apuntando a lo mismo en el constructor por copia
        nSupplies = l.nSupplies;
        nWeapons = l.nWeapons;
        nShields = l.nShields;
        nHangars = l.nHangars;
        nMedals = l.nMedals;
    }
    
    LootToUI getUIversion(){
        return (new LootToUI(this));
    }
    
    public String toString(){
        return getUIversion().toString();    
    }
    
    int getNSupplies() {
        return nSupplies;
    }
    
    int getNWeapons() {
        return nWeapons;
    }
    
    int getNShields() {
        return nShields;
    }
    
    int getNHangars() {
        return nHangars;
    }
    
    int getNMedals() {
        return nMedals;
    }
    
    boolean getEfficient() {
        return efficient;
    }
    
    boolean spaceCity() {
        return spaceCity;
    }
}
